pkg/oc/admin/diagnostics/diagnostics/cluster/scc.go

```diff
@@ -0,0 +1,95 @@
+package cluster
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	kerrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/apis/authorization"
+	authorizationtypedclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/authorization/internalversion"
+
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	"github.com/openshift/origin/pkg/oc/admin/diagnostics/diagnostics/types"
+	"github.com/openshift/origin/pkg/oc/admin/diagnostics/diagnostics/util"
+	policycmd "github.com/openshift/origin/pkg/oc/admin/policy"
+	securityapi "github.com/openshift/origin/pkg/security/apis/security"
+	securitytypedclient "github.com/openshift/origin/pkg/security/generated/internalclientset/typed/security/internalversion"
+)
+
+// ClusterSCC is a Diagnostic to check that the default securitycontextconstraints match expectations
+type ClusterSCC struct {
+	SCCClient securitytypedclient.SecurityContextConstraintsInterface
+	SARClient authorizationtypedclient.SelfSubjectAccessReviewsGetter
+}
+
+const SccName = "ClusterSCC"
+
+func (d *ClusterSCC) Name() string {
+	return SccName
+}
+
+func (d *ClusterSCC) Description() string {
+	return "Check that the default SecurityContextConstraints are present and contain the expected permissions"
+}
+
+func (d *ClusterSCC) Requirements() (client bool, host bool) {
+	return true, false
+}
+
+func (d *ClusterSCC) CanRun() (bool, error) {
+	if d.SCCClient == nil {
+		return false, fmt.Errorf("must have client.SubjectAccessReviews")
+	}
+
+	return util.UserCan(d.SARClient, &authorization.ResourceAttributes{
+		Verb:     "list",
+		Group:    securityapi.GroupName,
+		Resource: "securitycontextconstraints",
+	})
+}
+
+func (d *ClusterSCC) Check() types.DiagnosticResult {
+	r := types.NewDiagnosticResult(SccName)
+	reconcileOptions := &policycmd.ReconcileSCCOptions{
+		Confirmed:      false,
+		Union:          true,
+		Out:            ioutil.Discard,
+		SCCClient:      d.SCCClient,
+		InfraNamespace: bootstrappolicy.DefaultOpenShiftInfraNamespace,
+	}
+
+	changedSCCs, err := reconcileOptions.ChangedSCCs()
```

Comment:
Do we need to invoke `ReconcileSCCOptions.Validate()` to ensure that everything has been configured correctly?

Reply:
I think it is not necessary. `ReconcileSCCOptions.Validate()` https://github.com/openshift/origin/blob/12550a638027fb4b71332162f1d459a9d7d3c1c7/pkg/oc/admin/policy/reconcile_sccs.go#L140 is designed for the validation of command `oc adm policy reconcile-sccs`. If this diagnostics scc also called, it may cause some error in the future when `Validate()` was updated only for `oc adm policy reconcile-sccs`'s validation.

Reply:
On the other hand, if they add some member that will be used by `ChangedSCCs()`, we will notice that fact very late.

Reply:
I understand and agree with you. But `Validate()` in `reconcile-sccs.go` uses `NSClient`, which needs `clientcmd.Factory`. I am wondering that bringing `clientcmd.Factory` into here is not appropriate.

https://github.com/openshift/origin/blob/12550a638027fb4b71332162f1d459a9d7d3c1c7/pkg/oc/admin/policy/reconcile_sccs.go#L144-L146